### ISSUE TITLE
Cluster Tools is display incorrect version for legacy apps after upgrading rancher to 2.6.0

### ISSUE THREAD

**Issue by slickwarren** [rancher/dashboard#3825]
**Rancher Server Setup**
- Rancher version: v2.5.9 -> 2.6.0-rc2
- Installation option (Docker install/Helm Chart): helm
   - If Helm Chart, Kubernetes Cluster and version (RKE1, RKE2, k3s, EKS, etc): v1.20.9
- Proxy/Cert Details:

**Information about the Cluster**
- Kubernetes version: v1.20.9
- Cluster Type (Local/Downstream):downstream
   - If downstream, what type of cluster? (Custom/Imported or specify provider for Hosted/Infrastructure Provider): custom

**Describe the bug**

after upgrading rancher, user has the option of upgrading monitoring v1 to 0.3.0 (0.2.2 was the latest in rancher 2.5.9). However, `cluster tools` only displays the latest version as 0.2.2 (see additional context for more information)

**Additional context**

After some investigation, it appears that cluster tools is displaying the latest version for each app (not the currently installed version) this is a screenshot from the rancher v2.5.9 -> 2.6.0-rc2 cluster after the upgrade (stays the same after upgrading monitoring from 0.2.2 -> 0.3.0)
<img width="1791" alt="Screen Shot 2021-08-11 at 8 30 47 PM" src="https://user-images.githubusercontent.com/16691014/129134469-d69e7b12-7467-4f8a-920b-8a0ee84651db.png">

and here's the cluster tools version on a clean install of rancher v2.6.0-rc2:
<img width="539" alt="Screen Shot 2021-08-11 at 8 34 05 PM" src="https://user-images.githubusercontent.com/16691014/129134667-5d1b923d-9de6-4d1a-be9a-c4bb60155741.png">

**To Reproduce**
 
* install rancher v2.5.9
* add a downstream cluster, and install monitoring (v1) v0.2.2 (the latest version)
* upgrade rancher to v2.6.0
* go to cluster tools in the new UI

**Result**
* latest version displayed on cluster tools is not correct

**Expected Result**

monitoring v0.3.0 should display on 2.6.0, regardless of the rancher version you've upgraded from 

**Screenshots**

**Comment by nwmac** [comment]
@neillsom This might be a bit more involved - I recommend using bovine - https://github.com/nickgerace/bovine - to run a Rancher backend in docker locally - this will make it easy to upgrade - so you should be able to run 2.5.9 and then upgrade to 2.6.0 to reproduce.

I suspect the code in the cluster tools page that adds the fake cards for the legacy apps is assuming the versions are ordered with newest first, where maybe with an upgrade, this is not the case. First step is to reproduce the issue.

**Comment by slickwarren** [comment]
on rancher 2.6-head (78b25c3): I still see the same issue. Repro steps:
* deploy rancher v2.5.11
* deploy monitoring v0.2.2 on a downstream cluster
* upgrade to 2.6-head

* visit cluster-tools page for the downstream cluster

<img width="399" alt="Screen Shot 2021-11-15 at 3 50 01 PM" src="https://user-images.githubusercontent.com/16691014/141870764-48089df3-8e38-4c7b-8965-20a8eb79d9e6.png">
<img width="1783" alt="Screen Shot 2021-11-15 at 3 49 53 PM" src="https://user-images.githubusercontent.com/16691014/141870769-94d1eaf2-de48-4de3-858e-2887b16e6300.png">

**Comment by slickwarren** [comment]
I'm still seeing the issue on 2.6-head (da2e59c):
* deploy rancher 2.5.11
* deploy a downstream cluster and run monitoring (v1) v0.2.2
* upgrade rancher to 2.6.3-rc4
* go to the downstream cluster -> cluster tools


result:
* version displayed is 0.2.2. 


<img width="470" alt="Screen Shot 2021-12-07 at 1 40 27 PM" src="https://user-images.githubusercontent.com/16691014/145110400-bd29f0c1-227d-457e-8975-def62fe2f69f.png">

* Expecting cluster tools to display that monitoring v1 can be upgraded to 0.3.1. Similar to the screenshot below for alerting-drivers

<img width="454" alt="Screen Shot 2021-12-07 at 1 39 21 PM" src="https://user-images.githubusercontent.com/16691014/145110293-8b04efd5-34cb-47ac-b8c0-cf8d3e2065f0.png">

**Comment by davidnuzik** [comment]
**My checks PASSED**
I wasn't quite able to reproduce the issue. However, I did observe the change that shows the upgrade options now in the UI.
 
**Reproduction Environment:**

Rancher version: v2.5.11 -> v2.6.0-rc2
Rancher cluster type: HA
 
Downstream cluster type: RKE1
Downstream K8s version: v1.20
 
**Reproduction steps:**

1. Deploy Rancher 2.5.11
2. Deploy a downstream cluster (RKE1 in my case with EC2 Node driver)
3. Install monitoring
4. Upgrade Rancher to v2.6.0-rc2
5. Explore the downstream cluster, click on Cluster Tools in the left nav pane. I don't see the old version I see the new one (I didn't quite reproduce - I tried a couple additional times and couldn't repro).
![image](https://user-images.githubusercontent.com/45179589/146264872-111d6e83-ce4b-45dc-a769-ff332dcc77b3.png)
 
-----
 
**Validation Environment:**

Rancher version: v2.5.11 -> v2.6-rc5
Rancher cluster type: HA
 
Downstream cluster type: RKE1
Downstream K8s version: v1.20
 
**Validation steps:**

1. Execute the exact same steps as those outlined in the reproduction section above.
2. Notice that now the upgrade path is shown in the UI:

![image](https://user-images.githubusercontent.com/45179589/146269593-67208081-d49f-4d5d-bb18-5f403e68d099.png)
